homeassistant/components/zwave_js/api.py

```diff
@@ -266,6 +267,40 @@ def node_removed(event: dict) -> None:
     )
 
 
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/get_configuration_values",
+        vol.Required(ENTRY_ID): str,
+        vol.Required(NODE_ID): int,
+    }
+)
+@callback
+def websocket_get_configuration_values(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict
+) -> None:
+    """Get a list of configuration values for a Z-Wave node."""
+    entry_id = msg[ENTRY_ID]
+    node_id = msg[NODE_ID]
+    client = hass.data[DOMAIN][entry_id][DATA_CLIENT]
+    node = client.driver.controller.nodes[node_id]
+    values = node.get_configuration_values()
+    result = {}
+    for value in values:
+        result[value] = {
+            "property": values[value].data["property"],
```

Comment:
why not use the `.property`, `.property_name`, `.metadata`, and `.value` attributes directly?

Reply:
I would also include the `type` attribute in the result set. That will allow the frontend to show the right metadata based on the type of configuration value

Reply:
I wasn't able to get using .property directly to work:
```
  File "/home/cgarwood/home-assistant/core/homeassistant/components/zwave_js/api.py", line 291, in websocket_get_configuration_values
    "property": values[value].property,
AttributeError: 'ConfigurationValue' object has no attribute 'property'
```

`type` was included under the metadata key

Reply:
Property_ with underscore perhaps?

Reply:
yes it's `property_`, and the `type` under metadata and the `type` provided directly on the `ConfigurationValue` are different. Perhaps we should rename that latter `type` to `configuration_value_type` to avoid confusion

Reply:
https://github.com/home-assistant-libs/zwave-js-server-python/pull/118